vllm/v1/worker/gpu_model_runner.py

```diff
@@ -3177,13 +3184,87 @@ def _get_eagle3_aux_layers_from_config(self) -> tuple[int, ...] | None:
 
         return None
 
-    def reload_weights(self) -> None:
-        assert getattr(self, "model", None) is not None, (
-            "Cannot reload weights before model is loaded."
+    def reload_weights(
```

Comment:
so a lot of functionality is here, does this mean all runners have to be modified to include these?